### PR TITLE
Issues 2009 - Settings Import/Exporter

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR for issues #2009 

## How Has This Been Tested?
See the Video for Testing 
Video Link: http://screencast-o-matic.com/watch/cb6TrYIRtv

## Screenshots (jpeg or gifs if applicable):
Step 1:
image

Step 2: 
image

Step 3:
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/admin-actions.php [hunk 1]
```diff
@@ -568,13 +568,98 @@ function give_donation_import_callback() {
...
+	 *
+	 * @return array $url
+	 */
+	$json_data['url'] = give_import_page_url( (array) apply_filters( 'give_import_core_settings_success_url', array(
+		'step'          => ( empty( $json_data['success'] ) ? '1' : '3' ),
+		'importer-type' => 'import_core_setting',
+		'success'       => ( empty( $json_data['success'] ) ? '0' : '1' ),
+	) ) );
+
+	wp_die( json_encode( $json_data ) );
```
Comment (ravinderk) [inline includes/admin/admin-actions.php]: @raftaar1191 Why did not you use `wp_send_json` fx

Comment (raftaar1191) [reply includes/admin/admin-actions.php]: I don't know y I did that. I guess I just copy the code and past it right over here 

Fixed now @ravinderk for pointing this out to me

**Comment by DevinWalker** [comment]
## Review Findings
This is a nice looking addition and will help users port their settings over. 

### Support for Add-on's Settings
Many add-ons have global settings pages. The importer should support those configurations as well. For instance, Fee Recovery has settings: 



@raftaar1191 please ensure add-on settings are able to be imported/exported.

### Text Updates





Remove all mentions of "Core" settings since this will support add-on settings as well.


Change to:
> Please upload a valid JSON settings file.

**Comment by raftaar1191** [comment]
Checked and working on it @devinwalker

**Comment by raftaar1191** [comment]
@DevinWalker 

1>Many add-ons have global settings pages. The importer should support those configurations as well. For instance, Fee Recovery has settings:
-> Check with `Give - Gift Aid` and `Give - Tributes` it's working fine 

2> Text Updates
-> Done

3> Update Version to 1.8.17 Throughout
-> Done

**Comment by DevinWalker** [comment]
Works well after testing. Good work on this one @raftaar1191 👍

**Comment by raftaar1191** [comment]
Thanks @DevinWalker

**Pull request merged**